python/pyarrow/ipc.pxi

```diff
@@ -832,6 +880,27 @@ cdef class _RecordBatchFileWriter(_RecordBatchStreamWriter):
             self.writer = GetResultValue(
                 MakeFileWriter(c_sink, schema.sp_schema, self.options))
 
+_RecordBatchWithMetadata = namedtuple(
+    'RecordBatchWithMetadata',
+    ('batch', 'custom_metadata'))
+
+
+class RecordBatchWithMetadata(_RecordBatchWithMetadata):
```

Comment:
@pitrou just wondering, I know we have this pattern of having a second class subclassing the namedtuple elsewhere as well. But what is exactly the reason for this? (instead of using directly the namedtuple) To add a docstring?

Reply:
Yes, to add a docstring.